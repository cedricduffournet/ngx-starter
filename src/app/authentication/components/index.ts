export * from './auth-layout/auth-layout.component';
export * from './login-form/login-form.component';
export * from './register-form/register-form.component';
export * from './reset-password-form/reset-password-form.component';
export * from './send-reset-password-form/send-reset-password-form.component';
