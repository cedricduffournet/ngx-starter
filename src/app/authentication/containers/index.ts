export * from './login-view/login-view.component';
export * from './register-confirm-view/register-confirm-view.component';
export * from './register-view/register-view.component';
export * from './reset-password-view/reset-password-view.component';
export * from './send-reset-password-view/send-reset-password-view.component';
export * from './logout-modal/logout-modal.component';
