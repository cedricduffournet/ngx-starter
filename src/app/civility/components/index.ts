export * from './civility-add/civility-add.component';
export * from './civility-update/civility-update.component';
export * from './civility-delete/civility-delete.component';
export * from './civility-form/civility-form.component';
export * from './civility-item/civility-item.component';
export * from './civility-items/civility-items.component';
