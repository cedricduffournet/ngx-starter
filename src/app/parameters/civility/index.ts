export * from './parameter-civility.component';
