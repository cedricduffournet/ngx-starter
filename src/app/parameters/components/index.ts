export * from './referenced-table-menu/referenced-table-menu.component';
