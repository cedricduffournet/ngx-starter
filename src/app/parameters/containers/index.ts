export * from './list-parameters/list-parameters.component';
