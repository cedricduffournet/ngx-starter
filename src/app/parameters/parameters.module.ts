import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { TranslateModule } from '@ngx-translate/core';
import { PageInnerModule } from '@app/shared/page-inner/page-inner.module';
import { CivilityModule } from '@app/civility/civility.module';

import { ListParametersComponent } from '@app/parameters/containers';
import { ParameterCivilityComponent } from '@app/parameters/civility';
import { ReferencedTableMenuComponent } from '@app/parameters/components';

import { routing } from './parameters.routing';

@NgModule({
  imports: [
    CommonModule,
    TranslateModule,
    PageInnerModule,
    CivilityModule,
    routing,
    ReactiveFormsModule
  ],
  declarations: [
    ListParametersComponent,
    ReferencedTableMenuComponent,
    ParameterCivilityComponent
  ]
})
export class ParametersModule {}
