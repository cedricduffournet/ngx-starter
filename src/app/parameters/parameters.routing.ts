import { RouterModule, Routes } from '@angular/router';
import { ListParametersComponent } from '@app/parameters/containers';
import { ParameterCivilityComponent } from '@app/parameters/civility';
import { AuthGuard } from '@app/authentication/services';

export const routes: Routes = [
  {
    pathMatch: 'full',
    path: '',
    component: ListParametersComponent,
    data: { title: 'PARAMETERS' }
  },
  {
    path: 'civilities',
    component: ParameterCivilityComponent,
    canActivate: [AuthGuard],
    data: {
      roles: [
        'ROLE_CIVILITY_CREATE',
        'ROLE_CIVILITY_EDIT',
        'ROLE_CIVILITY_DELETE'
      ],
      title: 'CIVILITIES',
      previousLink: ['parameters']
    }
  }
];
export const routing = RouterModule.forChild(routes);
