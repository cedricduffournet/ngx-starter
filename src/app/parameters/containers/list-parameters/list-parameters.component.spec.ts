import { CommonModule } from '@angular/common';
import { TestBed, ComponentFixture } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';

import { provideMockStore } from '@ngrx/store/testing';

import { PageInnerModule } from '@app/shared/page-inner/page-inner.module';

import { ListParametersComponent } from '@app/parameters/containers';
import { ReferencedTableMenuComponent } from '@app/parameters/components';

describe('ParameterCivilityComponent', () => {
  let fixture: ComponentFixture<ListParametersComponent>;
  let component: ListParametersComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule, CommonModule, PageInnerModule],
      declarations: [ListParametersComponent, ReferencedTableMenuComponent],
      providers: [provideMockStore()]
    });

    fixture = TestBed.createComponent(ListParametersComponent);
    component = fixture.componentInstance;
  });

  it('should be created', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
