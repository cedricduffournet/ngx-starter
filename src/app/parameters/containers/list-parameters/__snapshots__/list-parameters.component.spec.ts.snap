// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParameterCivilityComponent should be created 1`] = `
<app-list-parameters
  canViewCivility$={[Function Store]}
  store={[Function MockStore]}
>
  <app-page-inner>
    <div
      class="px-3 py-3 "
    >
      <div
        class="page-section"
      >
        <div
          class="row"
        >
          
        </div>
      </div>
    </div>
  </app-page-inner>
</app-list-parameters>
`;
